### DISCUSSION TITLE
Error with union return type mixing pydantic schemas and response classes. "fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that issue.ExampleSchema | starlette.responses.JSONResponse is a valid Pydantic field type..."

### DISCUSSION THREAD

**Discussion by SquarishRectangle** [fastapi/fastapi#6020] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, status
from fastapi.responses import JSONResponse
from pydantic import BaseModel

app = FastAPI()


class ExampleSchema(BaseModel):
    foo: str
    bar: int


@app.post("/test")
async def test(thing: ExampleSchema) -> ExampleSchema | JSONResponse:
    if thing.foo == "bar":
        return JSONResponse(status_code=status.HTTP_400_BAD_REQUEST, content="foo is already bar")
    
    thing.foo = "bar"
    return thing
```

### Description

Full Error: "fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that issue.ExampleSchema | starlette.responses.JSONResponse is a valid Pydantic field type. If you are using a return type annotation that is not a valid 
Pydantic field (e.g. Union[Response, dict, None]) you can disable generating the response model from the type annotation with the path operation decorator parameter response_model=None. Read more: https://fastapi.tiangolo.com/tutorial/response-model/"

The error itself does present a fix that works, however, I don't want to disable doc generation entirely. Is there a way to present multiple possible return types that include both pydantic schemas and starlette responses?

### Operating System

Linux, Windows

### Operating System Details

Running Ubuntu through WSL. 

### FastAPI Version

0.89.1

### Python Version

3.10.9

### Additional Context

#5861 seems to fix a similar issue but with only one return type. The issue is different but perhaps a similar fix could be deployed?

**Comment by frankie567** [comment, marked as answer]
IMO, if your objective is just to raise an error, a better approach is to raise [`HTTPException`](https://fastapi.tiangolo.com/tutorial/handling-errors/#raise-an-httpexception-in-your-code):

```py
@app.post("/test")
async def test(thing: ExampleSchema) -> ExampleSchema:
    if thing.foo == "bar":
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="foo is already bar")
    
    thing.foo = "bar"
    return thing
```

You can even document errors so they appear in the automatic documentation (ref: https://fastapi.tiangolo.com/advanced/additional-responses/):

```py
@app.get(
    "/test",
    responses={
        400: {
            "content": {
                "application/json": {
                    "detail": "foo is already bar"
                }
            },
            "description": "The property `foo` already equals `bar`.",
        }
    },
)
async def test(thing: ExampleSchema) -> ExampleSchema:
    if thing.foo == "bar":
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="foo is already bar")
    
    thing.foo = "bar"
    return thing
```
